dashboard/app/controllers/registrations_controller.rb

```diff
@@ -21,17 +21,11 @@ class RegistrationsController < Devise::RegistrationsController
   # GET /users/sign_up
   #
   def new
-    session[:user_return_to] ||= params[:user_return_to]
-    if PartialRegistration.in_progress?(session)
-      user_params = params[:user] || ActionController::Parameters.new
-      user_params[:user_type] ||= session[:default_sign_up_user_type]
-      user_params[:email] ||= params[:email]
-      @user = User.new_with_session(user_params.permit(:user_type, :email), session)
-    else
-      save_default_sign_up_user_type
-      SignUpTracking.begin_sign_up_tracking(session, split_test: true)
-      super
+    sign_up_path = users_sign_up_account_type_url
+    if params[:user_return_to]
+      sign_up_path += "?user_return_to=#{params[:user_return_to]}"
```

Comment:
The `/users/account_type` page will save the `user_return_to` url param into `sessionStorage` so there's no more need for having a backend `session[:user_return_to]` variable to later reference in another route.